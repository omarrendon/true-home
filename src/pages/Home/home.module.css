.homeContainer {
  margin-top: 130px;
  width: 100%;
  height: auto;
}
.titleHome{
  text-align: center;
  font-size: 2rem;
}