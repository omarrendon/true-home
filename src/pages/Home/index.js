import { useEffect } from 'react';
import { SearchBar } from 'components/SearchBarFlight';
import { useSelector, useDispatch } from 'react-redux';
import { fetchFlights } from 'actions/fetchFlightsActions';
import style from './home.module.css';
import Modal from 'components/Modal';
import ScheduleList from 'components/ScheduleList';

const Home = () => {
  const dispatch = useDispatch();
  const { isLoading } = useSelector(state => state.flights);


  useEffect(() => {
    dispatch(fetchFlights());  
  }, []);

  const loadingModal = () => {
    return (
      <Modal
        showModal={isLoading}
        showClose={true}
        content={
          <div>
            <h4>Cargando...</h4>
          </div>
        }
      />
    )
  };
  return (
    <div className={style.homeContainer}>
      <h2 className={style.titleHome}>Selecciona tu vuelo</h2>
      <SearchBar>
        <SearchBar.Origin />
        <SearchBar.Destiny />
      </SearchBar>
      <ScheduleList />
      {loadingModal()}
    </div>
  );
};

export default Home;
