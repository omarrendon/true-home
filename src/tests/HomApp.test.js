import React from 'react';
import {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Home from 'pages/Home';


describe('test al componente <Home/>', () => {
  test('dbe dee mostrarse el componente Home', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot();
  });
});