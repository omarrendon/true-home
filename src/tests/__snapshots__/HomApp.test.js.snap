// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test al componente <Home/> dbe dee mostrarse el componente Home 1`] = `
<div
  className="homeContainer"
>
  <h2
    className="titleHome"
  >
    Selecciona tu vuelo
  </h2>
  <SearchBarFlight>
    <Origin />
    <Destiny />
  </SearchBarFlight>
  <ScheduleList />
</div>
`;
